solr/core/src/java/org/apache/solr/cloud/api/collections/BackupCmd.java

```diff
@@ -289,7 +294,10 @@ private void incrementalCopyIndexFiles(URI backupUri, String collectionName, ZkN
       NamedList<?> shardResp = (NamedList<?>)((NamedList<?>)shards.getVal(i)).get("response");
       if (shardResp == null)
         continue;
-      Integer shardIndexFileCount = (Integer) shardResp.get("indexFileCount");
+      // indexFileCount is expected to be found in case of incremental backups
+      // fileCount is expected to be found in case of snapshot backups
+      Integer shardIndexFileCount = (Integer) Optional.<Object> ofNullable(shardResp.get("indexFileCount"))
+          .orElse(shardResp.get("fileCount"));
```

Comment:
```suggestion
      Integer shardIndexFileCount = (Integer) shardResp.get(incremental ? "indexFileCount" : "fileCount");
```

Reply:
Thank you, @cpoerschke ! But it seems it is not relevant anymore, I've added field `indexFileCount` to duplicate field `fileCount` in shard snapshot response if you don't mind:

```java
      details.add("fileCount", files.size()); // TODO: for removal, replaced with indexFileCount
      details.add("indexFileCount", files.size());
      ...
      details.add("snapshotCompletedAt", Instant.now().toString()); // TODO: for removal, replaced with endTime
      details.add("endTime", Instant.now().toString());
```